runtime/ops/net.rs

```diff
@@ -267,7 +278,10 @@ async fn op_connect(
           .borrow::<Permissions>()
           .check_net(&args.hostname, args.port)?;
       }
-      let addr = resolve_addr(&args.hostname, args.port).await?;
+      let addr = resolve_addr(&args.hostname, args.port)
+        .await?
+        .next()
+        .ok_or_else(|| generic_error("No resolved address found"))?;
```

Comment:
Seems like unrelated change, do you need to change `resolve_addr` and `resolve_addr_sync` implementations in this PR?

Reply:
Not really, we can revert them, while these changes can be considered as small refactoring. That is, the changes about `resolve_addr` and `resolve_addr_sync` in this PR are the following:
- to let `resolve_addr` and `resolve_addr_sync` return iterators of `SocketAddr`, not just a single `SocketAddr`. These functions previously called `.next()` on an iterator internally, and returned it, so the callers of these functions had no way of handling other items than the first item in the iterator.  This change allows the caller to decide how to use the returned iterator.
- to use `generic_error` in deno_core instead of `anyhow::Context`. When I worked on #8743, I didn't know there were utility error functions like `generic_error`. 

Would it be better to revert it, or extract it and open a new PR?

Reply:
@magurotuna if that's not a big deal maybe extract to separate PR?

Reply:
@bartlomieju sure!

Reply:
#8891